controller.rst

```diff
@@ -348,18 +348,17 @@ Symfony provides a nice session object that you can use to store information
 about the user between requests. By default, Symfony stores the attributes in a
 cookie by using native PHP sessions.
 
-To retrieve the session, call
-:method:`Symfony\\Bundle\\FrameworkBundle\\Controller\\Controller::getSession`
-method on the ``Request`` object. This method returns a
-:class:`Symfony\\Component\\HttpFoundation\\Session\\SessionInterface` with easy
-methods for storing and fetching things from the session::
 
-    use Symfony\Component\HttpFoundation\Request;
+.. versionadded:: 3.3
+    The ability to request a ``Session`` in actions was introduced in Symfony 3.3.
 
-    public function indexAction(Request $request)
-    {
-        $session = $request->getSession();
+To retrieve the session, add the :class:`Symfony\\Component\\HttpFoundation\\Session\\Session`
```

Comment:
What about:
"To retrieve the session, type hint a controller argument with :class:`Symfony\\Component\\HttpFoundation\\Session\\SessionInterface` or any implementation:"
?
Then keep the example below which is much more useful to get access to method not available through the interface.